doc/sphinx/proof-engine/tactics.rst

```diff
@@ -3406,129 +3406,139 @@ Automation
 
-.. tacv:: debug auto
-   :name: debug auto
+   .. tacv:: debug auto
+      :name: debug auto
 
-   Behaves like :tacn:`auto` but shows the tactics it tries to solve the goal,
-   including failing paths.
+      Behaves like :tacn:`auto` but shows the tactics it tries to solve the goal,
+      including failing paths.
 
-.. tacv:: {? info_}auto {? @num} {? using {+ @lemma}} {? with {+ @ident}}
+   .. tacv:: {? info_}auto {? @num} {? using {+ @lemma}} {? with {+ @ident}}
 
-  This is the most general form, combining the various options.
+      This is the most general form, combining the various options.
 
 .. tacv:: trivial
    :name: trivial
 
-   This tactic is a restriction of auto that is not recursive
+   This tactic is a restriction of :tacn:`auto` that is not recursive
    and tries only hints that cost `0`. Typically it solves trivial
    equalities like :g:`X=X`.
 
-.. tacv:: trivial with {+ @ident}
-   :undocumented:
-
-.. tacv:: trivial with *
-   :undocumented:
-
-.. tacv:: trivial using {+ @lemma}
-   :undocumented:
-
-.. tacv:: debug trivial
-   :name: debug trivial
-   :undocumented:
-
-.. tacv:: info_trivial
-   :name: info_trivial
-   :undocumented:
-
-.. tacv:: {? info_}trivial {? using {+ @lemma}} {? with {+ @ident}}
-   :undocumented:
+   .. tacv:: trivial with {+ @ident}
+             trivial with *
+             trivial using {+ @lemma}
+             debug trivial
+             info_trivial
+             {? info_}trivial {? using {+ @lemma}} {? with {+ @ident}}
+      :name: _; _; _; debug trivial; info_trivial; _
+      :undocumented:
 
 .. note::
-  :tacn:`auto` either solves completely the goal or else leaves it
-  intact. :tacn:`auto` and :tacn:`trivial` never fail.
-
-The following options enable printing of informative or debug information for
-the :tacn:`auto` and :tacn:`trivial` tactics:
+  :tacn:`auto` and :tacn:`trivial` either solve completely the goal or
+  else succeed without changing the goal. Use :g:`solve [ auto ]` and
+  :g:`solve [ trivial ]` if you would prefer these tactics to fail when
+  they do not manage to solve the goal.
 
 .. flag:: Info Auto
           Debug Auto
           Info Trivial
           Debug Trivial
-   :undocumented:
 
-.. seealso:: :ref:`The Hints Databases for auto and eauto <thehintsdatabasesforautoandeauto>`
+   These options enable printing of informative or debug information for
+   the :tacn:`auto` and :tacn:`trivial` tactics.
 
 .. tacn:: eauto
    :name: eauto
 
    This tactic generalizes :tacn:`auto`. While :tacn:`auto` does not try
    resolution hints which would leave existential variables in the goal,
-   :tacn:`eauto` does try them (informally speaking, it usessimple :tacn:`eapply`
-   where :tacn:`auto` uses simple :tacn:`apply`). As a consequence, :tacn:`eauto`
+   :tacn:`eauto` does try them (informally speaking, it uses :tacn:`simple eapply`
+   where :tacn:`auto` uses :tacn:`simple apply`). As a consequence, :tacn:`eauto`
```

Comment:
You wrote earlier that `auto` uses something close to `simple apply`, but here it sounds like it uses exactly that.

Reply:
Indeed, I didn't want to change this text so much, but since you are asking, I'll update this.

Reply:
Fixed.